scripts/zkai_native_attention_mlp_adapter_compression_ablation_gate.py

```
@@ -0,0 +1,712 @@
+    gate = _dict(sources["gate"], "current single proof gate")
+    summary = _dict(gate.get("summary"), "current single proof summary")
+    routes = _dict(gate.get("routes"), "current single proof routes")
+    adapter_route = _dict(routes.get("adapter_boundary"), "adapter route")
+    native_route = _dict(routes.get("native_single_proof_object"), "native single proof route")
+    for key, expected in CURRENT_EXPECTED.items():
+        if key in {"proof_backend_version", "statement_version"}:
+            actual = native_route.get(key)
+        else:
+            actual = summary.get(key)
+        if actual != expected:
+            raise AdapterCompressionAblationError(f"current source {key} drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)